botocore/endpoint.py

```diff
@@ -121,6 +121,12 @@ def make_request(self, operation_model, request_dict):
     def create_request(self, params, operation_model=None):
         request = create_request_object(params)
         if operation_model:
+            if operation_model.request_compression is not None:
+                compress_aws_request(
+                    params['context']['client_config'],
```

Comment:
It shouldn't ever be in practice, but maybe should check if config is None here before performing compression.

Reply:
Do we have any tests that assert (or assume) that client_config is set here? If so, no need to worry about the check. Otherwise, I would add it.